acceptance/tests/vault/vault_tls_auto_reload_test.go

```diff
@@ -0,0 +1,182 @@
+	_, httpsAddress := consulCluster.SetupConsulClient(t, true)
+	rpcAddress := consulCluster.CreatePortForwardTunnel(t, 8300)
+
+	// here we can verify that the cert expiry changed
+	err, httpsCert := getCertificate(t, httpsAddress)
+	require.NoError(t, err)
+	logger.Logf(t, "HTTPS expiry: %s \n", httpsCert.NotAfter.String())
+
+	err, rpcCert := getCertificate(t, rpcAddress)
+	require.NoError(t, err)
+	logger.Logf(t, "RPC expiry: %s \n", rpcCert.NotAfter.String())
+
+	// Validate that consul sever is running correctly and the consul members command works
+	logger.Log(t, "Confirming that we can run Consul commands when exec'ing into server container")
+	membersOutput, err := k8s.RunKubectlAndGetOutputWithLoggerE(t, ctx.KubectlOptions(t), terratestLogger.Discard, "exec", fmt.Sprintf("%s-consul-server-0", consulReleaseName), "-c", "consul", "--", "sh", "-c", fmt.Sprintf("CONSUL_HTTP_TOKEN=%s consul members", bootstrapToken))
+	logger.Logf(t, "Members: \n%s", membersOutput)
+	require.NoError(t, err)
+	require.Contains(t, membersOutput, fmt.Sprintf("%s-consul-server-0", consulReleaseName))
+
+	// Deploy two services and check that they can talk to each other.
+	logger.Log(t, "creating static-server and static-client deployments")
+	k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-server-inject")
+	if cfg.EnableTransparentProxy {
+		k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-tproxy")
+	} else {
+		k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-inject")
+	}
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+		k8s.KubectlDeleteK(t, ctx.KubectlOptions(t), "../fixtures/bases/intention")
+	})
+	k8s.KubectlApplyK(t, ctx.KubectlOptions(t), "../fixtures/bases/intention")
+
+	logger.Log(t, "checking that connection is successful")
+	if cfg.EnableTransparentProxy {
+		k8s.CheckStaticServerConnectionSuccessful(t, ctx.KubectlOptions(t), staticClientName, "http://static-server")
+	} else {
+		k8s.CheckStaticServerConnectionSuccessful(t, ctx.KubectlOptions(t), staticClientName, "http://localhost:1234")
+	}
+
+	logger.Log(t, "Wait 30 seconds for certificates to rotate....")
+	time.Sleep(30 * time.Second)
+
+	err, httpsCert2 := getCertificate(t, httpsAddress)
+	require.NoError(t, err)
+	logger.Logf(t, "HTTPS 2 expiry: %s \n", httpsCert2.NotAfter.String())
+
+	err, rpcCert2 := getCertificate(t, rpcAddress)
+	require.NoError(t, err)
+	logger.Logf(t, "RPC 2 expiry: %s \n", rpcCert2.NotAfter.String())
+
+	// verify that a previous cert expired and that a new one has been issued
+	// by comparing the NotAfter on the two certs.
+	require.NotEqual(t, httpsCert.NotAfter, httpsCert2.NotAfter)
+	require.NotEqual(t, rpcCert.NotAfter, rpcCert2.NotAfter)
+
+}
+
+func getCertificate(t *testing.T, address string) (error, *x509.Certificate) {
```

Comment:
I think the convention is to have the `error` be the last returned value. Here, `(*x509.Certificate, error)` instead.